utils/bazel/llvm-project-overlay/libc/BUILD.bazel

```diff
@@ -62,7 +62,17 @@ config_setting(
     flag_values = {":mpfr": "system"},
 )
 
-############################## Support libraries #############################
+################################# Include Files ################################
+
+libc_support_library(
+    name = "internal_includes",
```

Comment:
All of the libc targets are split into separate targets and now you're making one big one to merge them all together. Can we either get rid of this or get rid of all the small ones?

Reply:
could you explain your issue in a little more detail? This support library globs all the macro and type headers that are available to the public headers, and since these are header-only internal implementation details I feel it's not a problem. Meanwhile the individual functions do need their own targets since a target can specify which specific list they want to build.

Reply:
I don't understand why all the other internal headers are factored into sublibraries but this one is not. It also overlaps with other libraries, which isn't will cause the build to turn messier over time as tooling no longer works.

The libc BUILD is very high maintenance and I thought there was a reason why it's set up that way. If it actually doesn't matter why not glob more things into one library and save a lot of work?

Reply:
Ah, I see what you mean by this overlapping with other libraries, I'll see about un-globbing the macro headers.

What do you mean by the libc build being high maintenance? Which sections would you suggest making a glob?

Reply:
with the current setup anny include change in libc requires a BUILD file change. It's the largest contributor to bazel build breakages, like right now it's broken by c63a291c64852a086cb447c7078e5312aac1a05c

Globbing all the __support things would certainly help, but I don't know what that means.